contracts/Vault1.sol

```diff
@@ -1,6 +1,49 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
 contract Vault1 {
-    
+    event Depositted(address by, uint256 amount);
+    event Withdrawn(address by, uint256 amount);
+
+    IERC20 public tokenContract;
+
+    mapping(address => uint256) public balanceOf;
+
+    constructor(address _tokenAddress) {
+        tokenContract = IERC20(_tokenAddress);
+    }
+
+    modifier noneZeroAmount(uint256 _amount) {
+        require(_amount > 0, "Amount should be greater than 0");
+        _;
+    }
+
+    function deposit(uint256 _amount) external payable noneZeroAmount(_amount) {
+        uint256 allowance = tokenContract.allowance(msg.sender, address(this));
+        require(allowance >= _amount, "Not enough allowance");
```

Comment:
This check is automatically being done in base ERC20 contract, you can skip this - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L163